src/components/tooltip/tooltip.js

```diff
@@ -187,6 +187,9 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
     }
 
     function setVisible (value) {
+      if ((setVisible.queued && setVisible.visible === !!value) || (!setVisible.queued && scope.visible === !!value)) {
```

Comment:
Nope because `setVisible.visible` can be still undefined and `scope.visible` is directive param that can be set from outside. There are 2 way this goes down:
1. `setVisible.queued` is set which means that someone already called `setVisible` and the change on `scope.visible` is pending. While `scope.visible` is not yet set correctly ($timeout not resolved), `setVisible.visible` is already set to the value that `scope.visible` will be after the $timeout resolves hence if `setVisible.visible === !!value` then we don't have to do nothing here.
2. `setVisible.queued` is not set, which means that `scope.visible` has the most current view value, hence we should compare the value passed to the function with `scope.value`